add buttons to header

varenya27 opened the pull request:
Removed the navbar menu for desktop view and added the buttons to header; mobile view remains the same

varenya27 committed:
add buttons to header

tarun-menta commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +53,85 @@ const useStyles = makeStyles((theme) => ({
+        </ListItem>
+      </List>
+ 
+      {/* <List>
+        <ListItem>
+          <FormControlLabel
+            control={<Switch onChange={toggleTheme} />}
+            // label="Toggle Theme"
+          />
+        </ListItem>
+      </List> */}
```
Delete comments like this

tarun-menta commented on src/components/NavbarDrawer.js:
```diff
@@ -134,7 +216,7 @@ function NavbarDrawer({ updateTT, toggleTheme }) {
             }}
           >
-            {drawer}
+        {drawerlist}
```
Indentation issues like this need to be fixed, ESLint should point all of them out to you

tarun-menta reviewed (changes requested):
The functionality on PC looks fine to me. However, ESLint is throwing a lot of errors/warnings, mainly formatting related - The major ones are inconsistent indentation, lack of spacing between operators, and unused variables. Please fix the ones that show up in the terminal when you run the app locally, you can ignore the PropTypes errors, those are due to some bad ESLint rules 

@varenya27

varenya27 committed:
made header buttons pretty, added hover messages, eslint stuff

varenya27 committed:
Delete settings.json

tarun-menta reviewed (changes requested):
Functionality is all good, and the buttons look much better. 

One tiny change - The descriptions in the tooltips need to be formatted better. Replace 'Toggle theme' with 'Toggle Theme', and 'Sync with aims timetable' with 'Sync with AIMS Timetable'

varenya27 committed:
capitalization

tarun-menta reviewed (approved):
All functionality is working, and the UI changes have been addressed

butter-chicken27 commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+      setIsMobile(window.innerWidth <= 500);
+    });
+  }, []);
+  const drawer1 = (
```
stop using variable names that dont convey any meaning to the readers. This is unacceptable.

butter-chicken27 commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+  const drawer1 = (
+    <div>
+      <Box display="flex" flexGrow={1}>
+        <IconButton
```
IconButtons don't inherit colour here and are out of sync with the text on the navbar. Add color="inherit"

butter-chicken27 commented on src/components/NavbarDrawer.js:
```diff
@@ -134,7 +197,7 @@ function NavbarDrawer({ updateTT, toggleTheme }) {
             }}
           >
-            {drawer}
+            {drawer2}
```
another crap variable name

butter-chicken27 commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
-  const handleDrawerToggle = () => {
+  function handleDrawerToggle() {
     setMobileOpen(!mobileOpen);
+  } // eslint-disable-next-line
```
currentab and setCurrentTab are not being used anywhere and could have just been removed instead of the overly liberal use of eslint-disable

tarun-menta reviewed (changes requested):
@varenya27 Please address the issues that @butter-chicken27 has pointed out

RachitKeertiDas commented on package.json:
```diff
@@ -17,13 +17,13 @@
     "lint-staged": "^10.4.0",
     "local-storage": "^2.0.0",
-    "moment": "^2.29.1",
+    "moment": "^2.29.2",
```
Please try not to make changes in package.json 

In any case, your changes do not really change the final resolved versions.

RachitKeertiDas commented on package.json:
```diff
@@ -17,13 +17,13 @@
     "react-firebase-hooks": "^2.2.0",
     "react-router-dom": "^5.2.0",
-    "react-scripts": "4.0.3",
+    "react-scripts": "^4.0.3",
```
Ditto.

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
   const classes = useStyles();
   const [mobileOpen, setMobileOpen] = React.useState(false);
-  const handleDrawerToggle = () => {
+  function handleDrawerToggle() {
```
Keep this as an arrow function.

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+    setCurrentTab(newTab);
   };
-  const drawer = (
+  useEffect(() => {
```
I think this `useEffect` hook is not doing anything. `currentTab` is not being used anywhere. This is what @butter-chicken27 was pointing out.

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -105,27 +163,32 @@ function NavbarDrawer({ updateTT, toggleTheme }) {
-            variant="temporary"
-            open={mobileOpen}
+            variant="temporary" // eslint-disable-next-line
+            open={mobileOpen} // eslint-disable-next-line
```
No need to use `eslint-disable-next-line here` please. Arrow Functions work.

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+            firebase.auth().signOut();
+            window.location.reload();
+          }}
+          onMouseOver
```
`onMouseOver` is probably expecting a function. 
I don't think you use `onMouseOver` anyway.
You can simply remove it.

Otherwise this is just spamming the console

varenya27 replied:
I was using onMouseOver for the tooltip thing (hovering on the button shows title). Should i look for a different way to do this?

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+        </IconButton>
+        <Box width={20} />
+        <IconButton
+          onMouseOver
```
Ditto as noted below.

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -105,27 +163,32 @@ function NavbarDrawer({ updateTT, toggleTheme }) {
+            <IconButton
+              color="inherit"
+              aria-label="Open drawer"
+              edge="start" // eslint-disable-next-line
```
no `// eslint-disable-next-line`

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -38,7 +40,7 @@ const useStyles = makeStyles((theme) => ({
   toolbar: theme.mixins.toolbar,
   drawerPaper: {
-    width: drawerWidth,
+    // width: drawerWidth,
```
does this change have any effect at all  ? Otherwise, I am open to removing this chunk completely.
@peppermenta @butter-chicken27

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+    <div>
+      <Box display="flex" flexGrow={1}>
+        <IconButton
+          padding={10}
```
This padding should not be applied to the mobile view.

varenya27 replied:
IconButtons aren't displayed in the mobile view anyway, so I don't think that would matter. I'll remove the padding for the title in mobile view.

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+        >
+          <SyncIcon />
+        </IconButton>
+        <Box width={20} />
```
Do not use empty `div` for spacing. Use margins on the `iconButton` instead

RachitKeertiDas commented on src/components/NavbarDrawer.js:
```diff
@@ -52,10 +54,66 @@ const useStyles = makeStyles((theme) => ({
+        >
+          <BrightnessHighIcon />
+        </IconButton>
+        <Box width={20} />
```
Do not use empty div for spacing. Use margins on the iconButton instead

RachitKeertiDas commented on src/components/NavBarDrawer.css:
```diff
@@ -1,3 +1,9 @@
 img {
   height: 1.2em;
 }
+button {
```
Please do not make changes targeting entire HTML tags.
This could have adverse effects elsewhere.

Also, I don't think we're depending on this anywhere, as buttons are rarely `display:flex` in our code. I could be wrong though

RachitKeertiDas commented on src/components/NavBarDrawer.css:
```diff
@@ -1,3 +1,9 @@
+button {
+  align-items: center;
+}
+.MuiListItemIcon-alignItemsFlexStart {
```
Is this class being used anywhere? I couldn't see an effect after removing this.

RachitKeertiDas reviewed (changes requested):
Hello,

This is a more detailed review of the code. Please also try to address these changes along with @butter-chicken27 's suggestions.

Thanks

varenya27 committed:
changed up variable names
added color inherit to icon buttons
capitalized the button titles in navbardrawer (for mobile view)
removed currentTab and setCurrentTab
reverted back to the original package.json
handleDrawerToggle is now an arrow function (as earlier), this fixed the remaining two eslint errors
removed onMouseOver from the icon buttons
used margin for spacing between icon buttons instead of an empty div box of some constant width
removed padding for title in mobile view
removed the unnecessary stuff form NavBarDrawer.css
